config/jobs/kubernetes-csi/csi-driver-host-path/csi-driver-host-path-config.yaml

```diff
@@ -0,0 +1,678 @@
+# generated by gen-jobs.sh, do not edit manually
+
+presubmits:
+  kubernetes-csi/csi-driver-host-path:
+  - name: pull-sig-storage-csi-driver-host-path-1-13-on-kubernetes-1-13
```

Comment:
Is that meant to save resources in the Prow test cluster?

We trade less resource usage against the risk that some breaking change slips through.

Reply:
When you are submitting a PR, you are only modifying the latest version of a sidecar/driver.  Your PR changes should not impact already tagged and released deployments.

Reply:
Testing against master is problematic, because then pre-submit jobs become non-deterministic: some change in K8S master can break the component and there's nothing can be done about it in the PR itself.

I've reduced the matrix to just two jobs: the latest stable release and the release before that.

Reply:
I'm still confused why we need to run 1.13 and 1.14 versions of the driver in pull jobs?  It should just be testing the latest build of the driver and not previous releases.

I think these are the only jobs we need (for pull jobs):
* driver-master-kubernetes-1.14
* driver-master-kubernetes-1.13

Reply:
Similar story for the sidecars.

Reply:
> I'm still confused why we need to run 1.13 and 1.14 versions of the driver in pull jobs?

This is still about `pull-sig-storage-csi-driver-host-path-1-13-on-kubernetes-1-13`, right?

> It should just be testing the latest build of the driver and not previous releases.

I think that's were the confusion comes from. That job *does* use the driver that was built as part of the PR job. If this needs to be more obvious, then I can add `CSI_PROW_BUILD_JOB: true` explicitly with a comment. It's left out at the moment because that's the default.

"1-13-on-kubernetes-1-13" just means that it then runs tests using the `deploy/kubernetes-1.13` deployment on Kubernetes 1.13 with the driver image replaced.

The deployment is relevant because it determines which sidecars to test with. The job then tests that the updated driver is still compatible with that older deployment, which is something that we are promising and thus need to test. I think this is a good candidate for a pre-submit job because it keeps everything else at a stable version.

I've been thinking a bit more about the situation where a new sidecar or driver feature depends on a new API in Kubernetes master. Right now, such changes can be submitted in a PR, but pre-submit tests will not cover them. Once merged, the periodic job with `on-kubernetes-master` will do that. If we want to enable pre-submit testing of such features, then we could do it like this:
- create a pr job which does not override the Kubernetes version
- bump the CSI_PROW_KUBERNETES_VERSION in the component to  some known-good revision of Kubernetes, potentially also switch to a newer version of the E2E tests
- submit the PR

I'm just not sure how important this is. My suggestion is to not add it at this time and wait whether there is a need.

Reply:
> The job then tests that the updated driver is still compatible with that older deployment, which is something that we are promising and thus need to test.

Hostpath driver compatibility is not something we're promising.  We are defining a versioning policy for sidecars, not for the hostpath driver.  That's why I think once we release a hostpath driver, we should not need to modify older deployments.